python_modules/dagster/dagster/_core/definitions/sensor_definition.py

```diff
@@ -181,6 +181,136 @@ def _get_partition_key_from_event_log_record(event_log_record: "EventLogRecord")
     return None
 
 
+def _get_asset_key_from_event_log_record(event_log_record: "EventLogRecord") -> AssetKey:
+    """
+    Given an event log record, returns the asset key for the event log record if it exists.
+
+    If the asset key does not exist, raises an error.
+    """
+    from dagster._core.storage.event_log.base import EventLogRecord
+
+    check.inst_param(event_log_record, "event_log_record", EventLogRecord)
+
+    dagster_event = event_log_record.event_log_entry.dagster_event
+
+    if dagster_event and dagster_event.asset_key:
+        return dagster_event.asset_key
+
+    raise DagsterInvariantViolationError(
+        "Asset key must exist in event log record passed into _get_asset_key_from_event_log_record"
+    )
+
+
+MAX_NUM_SKIPPED_EVENTS = 25
+
+
+class MultiAssetSensorContextCursor:
+    # Tracks the state of the cursor within the tick, created for utility purposes.
+    # Must call MultiAssetSensorEvaluationContext._update_cursor_after_evaluation at end of tick
+    # to serialize the cursor.
+    def __init__(self, cursor: Optional[str], context: "MultiAssetSensorEvaluationContext"):
+        self._unpacked_cursor: Dict[str, Tuple[Optional[str], Optional[int], List[int]]] = (
```

Comment:
What do you think about defining a `NamedTuple` class called something like `CursorAssetComponent` to house this information and make it clear what's included? Also, a name for this dictionary that communicates what the keys are would be helpful.